LanguageFeatures/Inline-classes/syntax_A02_t08.dart

```diff
@@ -0,0 +1,119 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion A rule for <inlineClassDeclaration> is added to the grammar,
+/// along with some rules for elements used in inline class declarations:
+///
+/// <inlineClassDeclaration> ::=
+///   'final'? 'inline' 'class' <typeIdentifier> <typeParameters>? <interfaces>?
+///   '{'
+///     (<metadata> <inlineMemberDeclaration>)*
+///   '}'
+///
+/// <inlineMemberDeclaration> ::= <classMemberDefinition>
+///
+/// @description Checks that it is a compile-time error to extend, implement (by
+/// non-inline class) or declare a mixin on a `final inline` class even in the
+/// same library where it is defined
+/// @author sgrekhov22@gmail.com
+
+// SharedOptions=--enable-experiment=inline-class
+
+final inline class FinalInlineClass {
+  final int x;
+  const FinalInlineClass([this.x = 0]);
+}
```

Comment:
It actually doesn't matter that the inline class is final: It's an error for any non-inline class/mixin/mixin-class to be a direct subtype of an inline class.

I think we ought to have a version of this test (and a version of the corresponding one using `implements`, and something similar for `on`) where the error arises just because the supertype is an inline class (not `final inline`), testing same-library and other-library.

Do we already have this?